client/src/context/AuthContext.tsx

```diff
@@ -0,0 +1,118 @@
+import React, {
+  FC,
+  createContext,
+  useCallback,
+  useState,
+  useMemo,
+} from 'react';
+import axios, { AxiosError } from 'axios';
+import {
+  AuthContextData,
+  User,
+  signupProps,
+  CustomErrorResponse,
+} from '../interfaces';
+
+export const AuthContext = createContext<AuthContextData>({
+  user: null,
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  login: async () => {},
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  signup: async () => {},
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  logout: async () => {},
+  errorMessage: '',
+});
+
+export const AuthProvider: FC<React.PropsWithChildren<object>> = ({
```

Comment:
the children on context provider is a RactNode 
you can create an interface for them like this : 

```
export interface MyContextProviderProps {
  children: ReactNode;
}```

And using it instead of the object :100: